velox/functions/prestosql/types/BigintEnumType.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <boost/algorithm/string.hpp>
+#include <map>
+
+#include "folly/CPortability.h"
+#include "velox/type/Type.h"
+
+namespace facebook::velox {
+
+class BigintEnumType : public BigintType {
+ public:
+  explicit BigintEnumType(
+      const std::string& enumName,
+      std::map<std::string, int64_t> enumMap)
+      : BigintType(/*providesCustomComparison*/ true),
+        enumName_(enumName),
+        enumMap_(std::move(enumMap)) {}
+
+  // Creates a BigintEnumType with the given name and enum map and inserts into
+  // the static map of instances.
+  static std::shared_ptr<const BigintEnumType> create(
```

Comment:
Why do we need a separate type registry just for these types? Why can't they be registered along with other types using registerCustomType from Type.h?